src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static nature() {
+    this.alive = this.alive.filter((dead) =>
+      dead.health !== 0);
+  }
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100, toHide = false) {
+    super(name, health);
+    this.hidden = toHide;
+
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+  bite(mammal) {
+    if (mammal.hidden === false) {
+      mammal.health -= 50;
+
+      if (mammal.health === 0) {
+        Animal.nature();
+      }
+    }
```

Comment:
[CODE LOGIC] The `bite` method should check whether the `mammal` is an instance of `Herbivore` and not hidden before applying damage. Currently, it only checks if `mammal.hidden` is `false`, which could lead to a bug if a `Carnivore` instance is passed to the method. Additionally, the method should not apply damage if the `mammal` is another `Carnivore`. You should add an `instanceof` check to make sure that the `mammal` is a `Herbivore` and not a `Carnivore`.